### PR TITLE
feat: resolve_import

### PR DESCRIPTION
**Pull request by dsherret**
Moves resolve_import from deno_graph:

https://github.com/denoland/deno_graph/blob/15db6e5fb6d3faea027e16c3d9ce6498b11beed2/src/module_specifier.rs#L53

This sort of makes sense to move here for situations where someone does not have an import map and still wants to resolve specifiers. This will be useful for code resuse in deno_graph and deno_config where deno_config does not depend on deno_graph.

### CODE REVIEW

File: rs-lib/src/lib.rs [hunk 1]
```diff
@@ -1118,6 +1119,8 @@ fn resolve_imports_match(
     return Ok(Some(url));
   }
 
+  log::debug!("Backtrace: {}", std::backtrace::Backtrace::force_capture());
```
Comment (bartlomieju) [inline rs-lib/src/lib.rs:1122]: Is this on purpose or left by accident?

**Review by bartlomieju — APPROVED** [review]

File: rs-lib/src/lib.rs (hunk 1 above)
Comment (dsherret) [reply rs-lib/src/lib.rs:1122]: Accident. I don’t remember writing this lol

Comment (dsherret) [inline rs-lib/src/lib.rs:1122]:
```suggestion
```

File: rs-lib/src/lib.rs [hunk 2]
```diff
@@ -1118,6 +1119,8 @@ fn resolve_imports_match(
...
+
```
Comment (dsherret) [inline rs-lib/src/lib.rs:1123]:
```suggestion
```

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**